File: JavaScript (Node.js & React)/JavaScript - INSY7314 - AtlasPay (Node.js & React)/payments-api/src/models/Payment.js
// src/models/Payment.js
const mongoose = require("mongoose");

const paymentSchema = new mongoose.Schema(
  {
    payerId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
    beneficiaryId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Beneficiary",
      required: true,
    },
    amount: { type: Number, required: true },
    currency: { type: String, required: true },
    provider: { type: String, default: "SWIFT" }, // hidden from customer
    status: {
      type: String,
      enum: ["pending", "verified", "declined"],
      default: "pending",
    },
    verifiedBy: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
    verifiedAt: { type: Date },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Payment", paymentSchema);

File: Kotlin/Kotlin - PROG7314 - EliteRentals (Mobile App & API Development)/Elite-Rentals-Rest-API/EliteRentalsAPI/Models/Payment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EliteRentalsAPI.Models
{
    public class Payment
    {
        [Key]
        public int PaymentId { get; set; }

        [ForeignKey("Tenant")]
        public int TenantId { get; set; }
        public User? Tenant { get; set; }

        public decimal Amount { get; set; }
        public DateTime Date { get; set; } = DateTime.UtcNow;
        public string Status { get; set; } = "Pending";
        public byte[]? ProofData { get; set; }
        public string? ProofType { get; set; }
    }
}
